### PR TITLE
fix for 1738

### PR DESCRIPTION
**Pull request by jennyf19**
#1738

### CODE REVIEW

**Review by bgavrilMS — APPROVED** [review]
LGTM, are there any unit tests that can be added? The way we choose the tenant id is quite subtle.

**Review by bgavrilMS — COMMENTED** [review]
Have some questions, happy to discuss offline.

File: src/Microsoft.Identity.Web/TokenAcquisition.cs [hunk 1]
```diff
@@ -852,11 +852,25 @@ private IConfidentialClientApplication BuildConfidentialClientApplication(Merged
                 if (builder != null)
                 {
                     builder.WithSendX5C(mergedOptions.SendX5C);
+
+                    ClaimsPrincipal? user = GetUserFromHttpContext();
```
[the remark this answers is not in the document]
Comment (jennyf19) [reply src/Microsoft.Identity.Web/TokenAcquisition.cs:856]: That is what is happening. The ClaimsPrincipal contains the result of the validated token.

[the remark this answers is not in the document]
Comment (jennyf19) [reply src/Microsoft.Identity.Web/TokenAcquisition.cs:856]: We have tests for GetTenantId(), for example:  GetTenantId_WithTidOrTenantIdClaims_ReturnsTenantId

Testing the entire flow will be a bit more challenging and I would need a multi-tenant web API.

**Pull request merged**